src/legacy/core_plugins/vis_type_vislib/public/vislib/components/tooltip/_hierarchical_tooltip_formatter.js

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import numeral from 'numeral';
+import { renderToStaticMarkup } from 'react-dom/server';
+
+import { FormattedMessage, I18nProvider } from '@kbn/i18n/react';
+
+import { collectBranch } from './_collect_branch';
+
+export function hierarchicalTooltipFormatter(metricFieldFormatter) {
+  return function({ datum }) {
+    // Collect the current leaf and parents into an array of values
+    const rows = collectBranch(datum);
+
+    // Map those values to what the tooltipSource.rows format.
+    _.forEachRight(rows, function(row) {
+      row.spacer = _.escape(_.repeat('&nbsp;', row.depth));
+
+      let percent;
+      if (row.item.percentOfGroup != null) {
+        percent = row.item.percentOfGroup;
+      }
+
+      row.metric = metricFieldFormatter ? metricFieldFormatter.convert(row.metric) : row.metric;
+
+      if (percent != null) {
+        row.metric += ' (' + numeral(percent).format('0.[00]%') + ')';
+      }
+
+      return row;
+    });
+
+    return renderToStaticMarkup(
+      <I18nProvider>
+        <table className="visTooltip__table">
+          <thead>
+            <tr className="eui-textLeft visTooltip__label">
+              <FormattedMessage
+                tagName="th"
+                scope="col"
+                id="visTypeVislib.vislib.tooltip.fieldLabel"
+                defaultMessage="field"
+              />
+
+              <FormattedMessage
+                tagName="th"
+                scope="col"
+                id="visTypeVislib.vislib.tooltip.valueLabel"
+                defaultMessage="value"
+              />
+              <th scope="col">{/* {metricCol.label} */}</th>
```

Comment:
Why is this commented out? Seems like this is used.

Reply:
Yeah I couldn't tell where it was coming from. It was in the angular template but there was no `metricCol` property that was set on the `$scope`. Is it possible to be coming from `$rootScope`?

https://github.com/elastic/kibana/blob/56482ab043a96a6be7ae2cc78693f0bb7fc20ff8/src/legacy/ui/public/vis/components/tooltip/_hierarchical_tooltip_formatter.js#L29-L64

Reply:
Seems like a leftover to me - it should be safe to remove it completely